graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/GraylogPreflightGeneratePeriodical.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.bootstrap.preflight;
+
+import org.graylog.security.certutil.cert.storage.CertMongoStorage;
+import org.graylog.security.certutil.csr.CsrSigner;
+import org.graylog.security.certutil.csr.storage.CsrMongoStorage;
+import org.graylog.security.certutil.csr.storage.CsrStorage;
+import org.graylog2.cluster.NodePreflightConfig;
+import org.graylog2.cluster.NodePreflightConfigService;
+import org.graylog2.plugin.periodical.Periodical;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+@Singleton
+public class GraylogPreflightGeneratePeriodical extends Periodical {
+    private static final Logger LOG = LoggerFactory.getLogger(GraylogPreflightGeneratePeriodical.class);
+
+    private final NodePreflightConfigService nodePreflightConfigService;
+
+    private final CsrStorage csrStorage;
+    private final CertMongoStorage certMongoStorage;
+
+    private String caKeystoreFilename = "datanode-ca.p12";
+    private Integer DEFAULT_VALIDITY = 90;
+    private static final String DEFAULT_PASSWORD = "admin";
+
+    @Inject
+    public GraylogPreflightGeneratePeriodical(final NodePreflightConfigService nodePreflightConfigService,
+                                              final CsrMongoStorage csrStorage,
+                                              final CertMongoStorage certMongoStorage) {
+        this.nodePreflightConfigService = nodePreflightConfigService;
+        this.csrStorage = csrStorage;
+        this.certMongoStorage = certMongoStorage;
+    }
+
+    @Override
+    public void doRun() {
+        LOG.debug("checking if there are configuration steps to take care of");
+        final Path caKeystorePath = Path.of(caKeystoreFilename);
+
+        try {
+            char[] password = DEFAULT_PASSWORD.toCharArray();
+            KeyStore caKeystore = KeyStore.getInstance("PKCS12");
+            caKeystore.load(new FileInputStream(caKeystorePath.toFile()), password);
+
+            var caPrivateKey = (PrivateKey) caKeystore.getKey("ca", password);
+            var caCertificate = (X509Certificate) caKeystore.getCertificate("ca");
```

Comment:
At some point we may want to configure "ca" alias, so that it is not a magic string here and inside org.graylog.security.certutil.ca.CACreator

Reply:
yes, we should make a note for the follow-up PR